src/Microsoft.OData.Core/UriParser/SemanticAst/SelectExpandClause.cs

```diff
@@ -76,18 +76,20 @@ internal void AddToSelectedItems(SelectItem itemToAdd, bool getOnlySubset = fals
         {
             ExceptionUtils.CheckArgumentNotNull(itemToAdd, "itemToAdd");
 
-            if (this.selectedItems.Any(x => x is WildcardSelectItem) && IsStructuralOrNavigationPropertySelectionItem(itemToAdd))
+            WildcardSelectItem wildcardSelectItem = this.selectedItems.FirstOrDefault(x => x is WildcardSelectItem) as WildcardSelectItem;
+            if (wildcardSelectItem != null && IsStructuralOrNavigationPropertySelectionItem(itemToAdd))
             {
+                wildcardSelectItem.AddSubsumed(itemToAdd);
                 return;
             }
 
-            bool isWildcard = itemToAdd is WildcardSelectItem;
+            wildcardSelectItem = itemToAdd as WildcardSelectItem;

```

Comment:
Why are we assigning `itemToAdd` to `wildcardSelectItem`?

Reply:
Because if `itemToAdd` is an instance of `WildcardSelectItem` we need access to it to add to the subsumed list. We could define a new variable (e.g. what @odero mentioned in one of his comments), but we already had a variable of type `WildcardSelectItem` defined so I figured I'd just use it again.